File: ClearAzureQueues/ViewModels/AccountViewModel.cs
using ClearAzureQueues.Models;
using System;
using System.Windows.Input;

namespace ClearAzureQueues.ViewModels {

    public class AccountViewModel {

        public AccountModel Model { get; set; }

        public ICommand Connect { get; set; }

        public AccountViewModel(Action onSuccess, Action onFailure) {
            Model = new AccountModel();
            Connect = new RelayCommand(
                x => !Model.IsConnecting,
                x => Model.Connect(onSuccess, onFailure));
        }
    }
}

File: ClearAzureQueues/ViewModels/AccountViewModel.fs
namespace ClearAzureQueues.ViewModels

open System
open System.Windows.Input

open ClearAzureQueues.Commands
open ClearAzureQueues.Models

type AccountViewModel(onSuccess, onFailure) =
    
    let model = new AccountModel()
    let connect =
        new RelayCommand(
            (fun _ -> not model.IsConnecting),
            (fun _ -> model.Connect(onSuccess, onFailure)))

    member public x.Model = model
    member public x.Connect = connect